.github/workflows/zephyr.yml

```diff
@@ -70,3 +69,134 @@ jobs:
         run: cd workspace && ./sof/zephyr/docker-run.sh
              ./sof/zephyr/docker-build.sh --cmake-args=-DEXTRA_CFLAGS=-Werror
              --cmake-args=--warn-uninitialized ${{ matrix.IPC_platforms }}
+
+  build-windows:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Using groups to avoid spamming the small results box with too
+        # many lines. Pay attention to COMMAS.
+        platforms: [
+          # - IPC3 default
+          imx8 imx8x imx8m,
+          tgl tgl-h,  # UNSUPPORTED! Will be removed
+          # - IPC4 default
+          mtl,
+          # Very few IPC3 platforms support IPC4 too.
+          -i IPC4 tgl tgl-h,
```

Comment:
I wonder if threre's a way to share the platform list with Windows? Not a huge thing though, can be handled later.

Reply:
Yeah, I've spend way too much time trying to implement this and finally gave up 😞 
Github right now does not allow to create a common matrix for few jobs.
What can be done (I've tried it), is to create a job A that needs job B to complete and passing input from one job to another. So the common matrix can be implemented like so:
1. create a "read-matrix" job that reads a matrix from json file
2. add build-windows and build-linux job that needs "read-matrix" job to complete and receives that matrix as job output
3. build-linux assigns received matrix as own job matrix filtering out windows array elements, same build-windows filtering out linux elements.

Unfortunately, parsing the matrix from json is very buggy - the moment where you read the json and try to assign received matrix as job matrix. Basically GH is picky about every single whitespace in the json file for some reason.
I had to remove all whitespaces (including space characters) from my json file and still there were issues parsing "-i IPC4 TGL" (since it has spaces inside).